_rules/printable-characters-shortcut-ffbc54.md

```diff
@@ -0,0 +1,575 @@
+---
+id: ffbc54
+name: No keyboard shortcuts use only printable characters
+rule_type: atomic
+description: |
+  This rule checks that if keyboard shortcuts are implemented using only printable characters, then there is a mechanism to disable the shortcut, or to remap the shortcut to use one or more non-printable characters keys, or the shortcut for a user interface component is only available when that component has focus.
+accessibility_requirements:
+  wcag21:2.1.4: # Character Key Shortcuts (A)
+    forConformance: true
+    failed: not satisfied
+    passed: satisfied
+    inapplicable: further testing needed
+acknowledgements:
+  authors:
+    - João Vicente
+    - Carlos Duarte
+---
+
+## Applicability
+
+The rule applies to any [keyboard event][]: 
+ - where the attribute `key` is a [printable character][] key; and
+ - the attribute `isComposing` is set to `false`; or the [keyboard event][] is part of an [composition event][] whose attribute `data` includes at least one [non-printable character][] key; and
+ - is [dispatched][] to an [event target][] within a [HTML document][].
+
+## Expectation
+
+For each test target:
+ - an [instrument][] to prevent the outcome of the [keyboard event][] is available; or
+ - if the [event target][] does not have [focus], the outcome of the event is prevented; or
```

Comment:
Same remark about "prevent". I think it needs more details.
For example, if the event as 2 outcomes, one when the target is focused and one when it is not, then we might say that the "outcome when focused" is prevented when the target does not have focus.

I guess you want to say something like "the event has no outcome" which probably needs to be detailed into something like "the event does not result in changes in the visual content"…

Of course, changing the visual content is not the only bad thing a shortcut can do. But that would only mean we pass bad cases, which is OK. And it is a start, maybe we can add more bad outcomes that need to be explicitly prevented (playing audio/video, redirecting the page, …)

Well, anyway I think we need to clarify "prevent the outcome", especially since "change the outcome" does prevent it (but replaces with another outcome which might be just as bad).